src/sagemaker/serve/builder/model_builder.py

```diff
@@ -577,6 +594,79 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
+    def _check_if_input_is_mlflow_model(self) -> bool:
+        """Checks whether an MLmodel file exists in the given directory.
+
+        Returns:
+            bool: True if the MLmodel file exists, False otherwise.
+        """
+        if self.inference_spec or self.model:
+            logger.info(
+                "Either inference spec or model is provided. "
+                "ModelBuilder is not handling MLflow model input"
+            )
+            return False
+
+        if not self.model_metadata:
+            logger.info(
+                "No ModelMetadata provided. ModelBuilder is not handling MLflow model input"
+            )
+            return False
+
+        path = self.model_metadata.get(MLFLOW_MODEL_PATH_MODEL_METADATA_ENTRY_KEY)
+        if not path:
+            logger.info(
+                "%s is not provided in ModelMetadata. ModelBuilder is not handling MLflow model "
+                "input",
+                MLFLOW_MODEL_PATH_MODEL_METADATA_ENTRY_KEY,
+            )
+            return False
+
+        mlmodel_file = "MLmodel"
+        # Check for S3 path
+        if path.startswith("s3://"):
+            s3_client = self.sagemaker_session.boto_session.client("s3")
```

Comment:
Can this be re-used instead?
https://github.com/aws/sagemaker-python-sdk/blob/48234aed26ab0edd885f6f0b11962c44ee448bd2/src/sagemaker/s3.py#L206

Reply:
Excellent point, we can reuse this